libr/bin/p/bin_art.c

```diff
@@ -63,19 +63,20 @@ static Sdb *get_sdb(RBinFile *bf) {
 	return ao? ao->kv: NULL;
 }
 
-static void *load_bytes(RBinFile *bf, const ut8 *buf, ut64 sz, ut64 la, Sdb *sdb){
+static bool load_bytes(void *bin_obj, RBinFile *bf, const ut8 *buf, ut64 sz, ut64 la, Sdb *sdb){
 	ArtObj *ao = R_NEW0 (ArtObj);
 	if (!ao) {
-		return NULL;
+		return false;
 	}
 	ao->kv = sdb_new0 ();
 	if (!ao->kv) {
 		free (ao);
-		return NULL;
+		return false;
 	}
 	art_header_load (&ao->art, bf->buf, ao->kv);
 	sdb_ns_set (sdb, "info", ao->kv);
-	return ao;
+	bin_obj = ao;
```

Comment:
what's the purpose of this line?

Reply:
The old function prototype returned a void* pointer to a bin_obj. I changed the prototype to return a bool, and the pointer is now returned in the first parameter of the function.

I thought about not adding that parameter and just use the bf parameter to set bf->o->bin_obj, as in most cases, that is the one pointer that is set by the function. I didn’t do it because I wasn’t sure whether there were other use cases that could differ.

This line was meant to keep the old behavior.

Reply:
please see my other comment about pointer uses. This line is not doing anything except changing the value of the *local variable* `bin_obj`, which is not reflected outside this function.

Reply:
This is all in response to issue #11553.

Reply:
I see, thanks! Yeah so either you add a `bool *loaded` to the arguments and you keep `void *` as return argument, or you make `bool` the return value and you add a `void **` as argument. I wouldn't set `o->bin_obj` directly in the `load_bytes` function because in some cases you may not need that.

Reply:
Added void ** parameter to load_bytes().